source/utils.ts

```diff
@@ -3,20 +3,50 @@ export type JSONObject = { [key: string]: JSON };
 
 type UnknownRecord = Record<string, unknown>;
 
-export function get(object: UnknownRecord, path: string) {
+const getTranslationWithCount = (
+  originalCount: number,
+  index: number,
+  translation: string,
+) => {
+  const options = translation.split(' | ');
+  const pluralized =
+    options[index >= options.length ? options.length - 1 : index];
+
+  if (pluralized.includes('{count}')) {
+    return pluralized.replace('{count}', `${originalCount}`);
+  }
+  return pluralized;
+};
+
+export function get(
+  object: UnknownRecord,
+  path: string,
+  count?: number,
+  pluralizationRules?: Record<string, (count: number) => number>,
+) {
   const keys = path.split('.');
   const last = keys.pop()!;
+  const locale = keys[0];
 
   let key: string | undefined;
   while ((key = keys.shift())) {
     if (typeof object !== 'object' || object === null) {
       break;
     }
-
     object = object[key] as UnknownRecord;
   }
 
-  return object?.[last];
+  const translation = object?.[last];
+
+  if (count !== undefined) {
+    let index = count;
+    if (pluralizationRules && pluralizationRules[locale]) {
+      index = pluralizationRules[locale](count);
+    }
+    return getTranslationWithCount(count, index, translation as string);
```

Comment:
Is this variable a string for sure? What if user ask for translation which isn't in the translation file?

Reply:
I haven't seen any case where it wasn't a string but I'll add a type check to be sure.